### PR TITLE
[DOCKER] docker file generating ubuntu18.04 and raspbian stretch installer

### PR DESCRIPTION
**Pull request by Mathiasb17**
Hi,

In the community a lot of people are linux newcomers and have a hard time making the wallet work on a wide variety of linux distributions.

I made a dockerfile that automatically generates an installer for ubuntu 18.04. I also added a manual for it (README.md). 

This way we could distribute the installer directly to users, without having them build the wallet from scratch. This is a lot easier as it only requires 4 easy linux command to install the wallet.

I also plan to add a dockerfile for raspberry pi, as a lot of people seem to have a hard time making it work, mainly because of their inexperience with linux.

What do you think ?

### CODE REVIEW

**Comment by Mathiasb17** [comment]
Note that this dockerfile should be exactly the same for raspbian, but i hadn't the chance to test it yet. I'm currently installing QEMU to do so.

File: tpay_docker/Ubuntu_18.04/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+     pkg-config zlib1g-dev libevent-dev \
+     bsdmainutils git libboost-all-dev \
+     libseccomp-dev libcap-dev libminiupnpc-dev \
+     libqt5gui5 libqt5core5a libqt5webkit5-dev \
+     libqt5dbus5 qttools5-dev qttools5-dev-tools \
+     libprotobuf-dev protobuf-compiler libqrencode-dev \
+     libssl-dev qt5-default
+RUN apt-get install -y wget
+
+#openssl
```
Comment (skafandri) [inline tpay_docker/Ubuntu_18.04/Dockerfile]: Is this necessary? Starting from Ubuntu 18.04, `libssl` comes with version `1.1.0`

Comment (Mathiasb17) [reply tpay_docker/Ubuntu_18.04/Dockerfile]: I wasn't aware of it, i will remove this and test it this evening, thanks.

**Comment by Mathiasb17** [comment]
I fixed the ubuntu 18.04 dockerfile accordingly to your remark.
I added the raspbian stretch dockerfile.

edit : ran both docker images this night, everything seems working fine, as well as the generated installers.

**Comment by Mathiasb17** [comment]
Hi,
I think a good way to distribute the install generated with the docker build is with an online shell script.
For the user it would be run like this :

```bash
curl http://tokenpay.com/install/ubuntu18_04/web_install.sh | sudo bash
```

For debian based distributions the content of the script would be : 

```bash
apt-get install -y build-essential libtool autotools-dev automake pkg-config zlib1g-dev libevent-dev bsdmainutils git libboost-all-dev libseccomp-dev libcap-dev libminiupnpc-dev libqt5gui5 libqt5core5a libqt5webkit5-dev libqt5dbus5 qttools5-dev qttools5-dev-tools libprotobuf-dev protobuf-compiler libqrencode-dev libssl-dev qt5-default wget

curl http://tokenpay.com/.../tokenpay > /usr/bin/tokenpay
curl http://tokenpay.com/.../tokenpayd > /usr/bin/tokenpayd
curl http://tokenpay.com/...tokenpay_icon.png > /usr/share/icons/tokenpay_icon.png
curl http://tokenpay.com/.../tokenpay.desktop > ~/.local/share/applications/tokenpay.desktop

mkdir ~/.tokenpay/
touch ~/.tokenpay/tokenpay.conf

chown -R $SUDO_USER ~/.tokenpay/
chown -R $SUDO_USER /usr/share/icons/tokenpay_icon.png
chown -R $SUDO_USER ~/.local/share/applications/tokenpay.desktop
chown -R $SUDO_USER /usr/bin/tokenpay
chown -R $SUDO_USER /usr/bin/tokenpayd
```
I tested it on a virtual machine with a local http server , seems to work fine.

A lot of applications use curl based commands to get installed. This is easy for everyone.

**Comment by skafandri** [comment]
Good job @Mathiasb17

**Pull request merged**